plugins/text_generation_webui_plugin.py

```diff
@@ -50,22 +55,22 @@ def streaming_request_http(self, query, user_id):
             "stream": True,
         }
 
-        chunks = []
-        ack_time = 0
-        first_token_time = 0
-        start_time = time.time()
+        result = RequestResult(user_id, query.get("text"), query.get("input_tokens"))
+
+        tokens = []
+        result.start_time = time.time()
 
-        # TODO add configurable timeout to requests
         response = requests.post(
-            self.route, headers=headers, json=data, verify=False, stream=True
+            self.host, headers=headers, json=data, verify=False, stream=True
         )
+
```

Comment:
We could also add `try...catch` here just the way you added it in the `hf_tgi_plugin`.

Reply:
Yes for sure, as mentioned in the PR description

> The other plugins definitely need error handling as well, but we can add this in subsequent PRs depending on which plugins we are using for our work.